services/motion/motion.go

```diff
@@ -16,12 +20,83 @@ import (
 func init() {
 	resource.RegisterAPI(API, resource.APIRegistration[Service]{
 		RPCServiceServerConstructor: NewRPCServiceServer,
-		RPCServiceHandler:           servicepb.RegisterMotionServiceHandlerFromEndpoint,
-		RPCServiceDesc:              &servicepb.MotionService_ServiceDesc,
+		RPCServiceHandler:           pb.RegisterMotionServiceHandlerFromEndpoint,
+		RPCServiceDesc:              &pb.MotionService_ServiceDesc,
 		RPCClient:                   NewClientFromConn,
 	})
 }
 
+// GetPlanRequest describes the request to the GetPlan interface method.
+// Contains the ComponentName the returned plan(s) should be associated with,
+// an optional  ExecutionID and an Extra parameter.
+type GetPlanRequest struct {
+	ComponentName resource.Name
+	LastPlanOnly  bool
```

Comment:
What does this do exactly? I can think of two or three potential behaviors.

Reply:
It makes it so that the get plan response only returns the most recent plan, rather than the full history of all replans which have happened for the execution in question. 

If the executionID is not provided, it defaults to the recent execution for the component. If the executionID it uses the execution of that id for that component.

Reply:
I've also added a comment to the type